amazon_cloud_code_generator/data/plugins/module_utils/core.py

```diff
@@ -249,19 +249,27 @@ def present(
 
         resource = None
 
-        for id in primary_identifier:
-            identifier[
-                snake_to_camel(id, capitalize_first=True)
-            ] = self.module.params.get(id)
+        if self.module.params.get("identifier"):
+            identifier = self.module.params.get("identifier")
+        else:
+            for id in primary_identifier:
+                identifier[
+                    snake_to_camel(id, capitalize_first=True)
+                ] = self.module.params.get(id)
+            identifier = json.dumps(identifier)
 
         try:
             resource = self.client.get_resource(
-                TypeName=type_name, Identifier=json.dumps(identifier)
+                TypeName=type_name, Identifier=identifier
             )
             results = self.update_resource(
                 resource, params, create_only_params, handlers
             )
         except self.client.exceptions.ResourceNotFoundException:
+            if self.module.params.get("identifier"):
+                self.module.fail_json(
+                    f"You must specify both {*primary_identifier, } to create a new resource. Option identifier must only be used to manipulate an existing resource."
```

Comment:
```suggestion
                    f"You must specify both {*primary_identifier, } to create a new resource. The identifier parameter can only be used to manipulate an existing resource."
```
The wording here was a little confusing.

Reply:
Done, thanks!